### PR TITLE
metrics: fixes for many golint warnings

### PR DESCRIPTION
**Pull request by kielbarry**
Fixed nearly all golint warnings not related to lack of documentation.

### CODE REVIEW

File: metrics/debug.go [hunk 1]
```diff
@@ -19,15 +19,15 @@ var (
 	gcStats debug.GCStats
 )
 
-// Capture new values for the Go garbage collector statistics exported in
+// CaptureDebugGCStats captures new values for the Go garbage collector statistics exported in
```
Comment (karalabe) [inline metrics/debug.go]: Please wrap documentation at around 80 characters. it looks very bad if you have a gigantic line longer than everything else (same below).

File: metrics/counter.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package metrics
 
 import "sync/atomic"
 
-// Counters hold an int64 value that can be incremented and decremented.
+// Counter methods hold an int64 value that can be incremented and decremented.
```
Comment (karalabe) [inline metrics/counter.go]: No need for `methods`... it's an interface, just drop the extra `s` and it's fine.

File: metrics/ewma.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"sync/atomic"
 )
 
-// EWMAs continuously calculate an exponentially-weighted moving average
+// EWMA methods continuously calculate an exponentially-weighted moving average
```
Comment (karalabe) [inline metrics/ewma.go]: It's an interface, not methods. `EWMA continuously calculates [...]` is enough

File: metrics/exp/exp.go [hunk 4]
```diff
@@ -1,5 +1,7 @@
-// Hook go-metrics into expvar
-// on any /debug/metrics request, load all vars from the registry into expvar, and execute regular expvar handler
+// Package exp hookes go-metrics into expvar
+// on any /debug/metrics request,
+// loads all vars from the registry into expvar,
+// and executes regular expvar handler.
```
Comment (karalabe) [inline metrics/exp/exp.go]:
```
// Package exp hooks go-metrics into expvar.
//
// On any /debug/metrics request, loads all vars from the registry into expvar,
// and executes regular expvar handler.
```

File: metrics/gauge.go [hunk 5]
```diff
@@ -2,7 +2,7 @@ package metrics
 
 import "sync/atomic"
 
-// Gauges hold an int64 value that can be set arbitrarily.
+// Gauge methods hold an int64 value that can be set arbitrarily.
```
Comment (karalabe) [inline metrics/gauge.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/gauge_float64.go [hunk 6]
```diff
@@ -2,7 +2,7 @@ package metrics
 
 import "sync"
 
-// GaugeFloat64s hold a float64 value that can be set arbitrarily.
+// GaugeFloat64 methods hold a float64 value that can be set arbitrarily.
```
Comment (karalabe) [inline metrics/gauge_float64.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/healthcheck.go [hunk 7]
```diff
@@ -1,6 +1,6 @@
 package metrics
 
-// Healthchecks hold an error value describing an arbitrary up/down status.
+// Healthcheck methods hold an error value describing an arbitrary up/down status.
```
Comment (karalabe) [inline metrics/healthcheck.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/histogram.go [hunk 8]
```diff
@@ -1,6 +1,6 @@
 package metrics
 
-// Histograms calculate distribution statistics from a series of int64 values.
+// Histogram methods calculate distribution statistics from a series of int64 values.
```
Comment (karalabe) [inline metrics/histogram.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/meter.go [hunk 9]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 )
 
-// Meters count events to produce exponentially-weighted moving average rates
+// Meter methods count events to produce exponentially-weighted moving average rates
```
Comment (karalabe) [inline metrics/meter.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/timer.go [hunk 10]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 )
 
-// Timers capture the duration and rate of events.
+// Timer methods capture the duration and rate of events.
```
Comment (karalabe) [inline metrics/timer.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/sample.go [hunk 11]
```diff
@@ -394,7 +395,7 @@ func SampleVariance(values []int64) float64 {
 	return sum / float64(len(values))
 }
 
-// A uniform sample using Vitter's Algorithm R.
+// UniformSample uniform sample using Vitter's Algorithm R.
```
Comment (karalabe) [inline metrics/sample.go]: // UniformSample represents a uniform sample using Vitter's Algorithm R.

File: metrics/sample.go [hunk 12]
```diff
@@ -10,7 +10,7 @@ import (
 
 const rescaleThreshold = time.Hour
 
-// Samples maintain a statistically-significant selection of values from
+// Sample methods maintain a statistically-significant selection of values from
```
Comment (karalabe) [inline metrics/sample.go]: Please don't add `methods` all over, it's a simple interface.

File: metrics/resetting_timer.go [hunk 13]
```diff
@@ -7,7 +7,7 @@ import (
 	"time"
 )
 
-// Initial slice capacity for the values stored in a ResettingTimer
+// InitialResettingTimerSliceCap sets the tnitial slice capacity for the values stored in a ResettingTimer
```
Comment (karalabe) [inline metrics/resetting_timer.go]: // InitialResettingTimerSliceCap is the initial slice capacity for the values
// stored in a ResettingTimer.

File: metrics/registry.go [hunk 14]
```diff
@@ -50,33 +50,33 @@ type Registry interface {
 	UnregisterAll()
 }
 
-// The standard implementation of a Registry is a mutex-protected map
+// StandardRegistry - the standard implementation of a Registry - is a mutex-protected map
```
Comment (karalabe) [inline metrics/registry.go]: // StandardRegistry is a mutex protected implementation of a Registry.

File: metrics/debug.go (hunk 1 above)
Comment (kielbarry) [reply metrics/debug.go]: Thank you - updated.

File: metrics/counter.go (hunk 2 above)
Comment (kielbarry) [reply metrics/counter.go]: Thank you - updated.

File: metrics/ewma.go (hunk 3 above)
Comment (kielbarry) [reply metrics/ewma.go]: Thank you - updated.

File: metrics/exp/exp.go (hunk 4 above)
Comment (kielbarry) [reply metrics/exp/exp.go]: Thank you - updated.

File: metrics/gauge.go (hunk 5 above)
Comment (kielbarry) [reply metrics/gauge.go]: Thank you - updated.

File: metrics/gauge_float64.go (hunk 6 above)
Comment (kielbarry) [reply metrics/gauge_float64.go]: Thank you - updated.

File: metrics/healthcheck.go (hunk 7 above)
Comment (kielbarry) [reply metrics/healthcheck.go]: Thank you - updated.

File: metrics/histogram.go (hunk 8 above)
Comment (kielbarry) [reply metrics/histogram.go]: Thank you - updated.

File: metrics/meter.go (hunk 9 above)
Comment (kielbarry) [reply metrics/meter.go]: Thank you - updated.

File: metrics/timer.go (hunk 10 above)
Comment (kielbarry) [reply metrics/timer.go]: Thank you - updated.

File: metrics/sample.go (hunk 11 above)
Comment (kielbarry) [reply metrics/sample.go]: Thank you - updated.

File: metrics/sample.go (hunk 12 above)
Comment (kielbarry) [reply metrics/sample.go]: Thank you - updated.

File: metrics/resetting_timer.go (hunk 13 above)
Comment (kielbarry) [reply metrics/resetting_timer.go]: Thank you - updated.

**Comment by fjl** [comment]
Sorry, this is stale now, closing. Thank you for your effort.

**Pull request closed without merging**